python/feathub/feature_views/feature_view.py

```diff
@@ -133,25 +137,35 @@ def get_output_fields(self, source_fields: List[str]) -> List[str]:
             raise FeathubException(
                 "Build this feature view before getting output fields."
             )
-        output_fields = []
+
+        feature_fields = [feature.name for feature in self.get_resolved_features()]
```

Comment:
Would the following code be shorter and a bit more readable?

```
feature_fields = [feature.name for feature in self.get_resolved_features()]
if self.keep_source_fields:
    extra_fields = list(set(source_fields) - set(feature_fields))
else:
    timestamp_and_keys = []
    for feature in self.get_resolved_features():
        if feature.keys is not None:
            timestamp_and_keys.extend(feature.keys)
    if self.timestamp_field is not None:
        timestamp_and_keys.append(self.timestamp_field)
    extra_fields = list(set(timestamp_and_keys) - set(feature_fields))
# Order extra fields in the same order as their position in source.
extra_fields = [
    f for f in source_fields if f in extra_fields
]
return extra_fields + feature_fields
```

Reply:
Yes, it's simpler. But the `set` doesn't keep order, but we want the `extra_fields` in the order as they appear in the source fields.

Reply:
Would the following code snippet ensure that the extra_fields have the same order as `source_fields`?

```
# Order extra fields in the same order as their position in source.
extra_fields = [
    f for f in source_fields if f in extra_fields
]
```

Reply:
Yes, you're right, I lost it.